src/silentpayments.rs

```diff
@@ -0,0 +1,546 @@
+
+        let (ffi_taproot_seckeys, n_taproot_seckeys) = match taproot_seckeys {
+            Some(keys) => {
+                let ffi_keys: &[*const ffi::Keypair] = transmute::<&[&Keypair], &[*const ffi::Keypair]>(taproot_seckeys.unwrap());
+                (ffi_keys.as_c_ptr(), keys.len())
+            }
+            None => (core::ptr::null(), 0),
+        };
+
+        let (ffi_plain_seckeys, n_plain_seckeys) = match plain_seckeys {
+            Some(keys) => {
+                let ffi_keys: &[*const u8] = transmute::<&[&SecretKey], &[*const u8]>(plain_seckeys.unwrap());
+                (ffi_keys.as_c_ptr(), keys.len())
+            }
+            None => (core::ptr::null(), 0),
+        };
+
+        let res = secp256k1_silentpayments_sender_create_outputs(
+            cx,
+            out_pubkeys_ptrs.as_mut_ptr(),
+            ffi_recipients_ptrs.as_mut_ptr(),
+            n_tx_outputs,
+            smallest_outpoint.as_ptr(),
+            ffi_taproot_seckeys,
+            n_taproot_seckeys,
+            ffi_plain_seckeys,
+            n_plain_seckeys,
+        );
+
+        if res == 1 {
+            Ok(out_pubkeys.into_iter().map(XOnlyPublicKey::from).collect())
+        } else {
+            Err(SenderOutputCreationError::Failure)
+        }
+    }
+}
+
+/// Struct to store label tweak result
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct LabelTweakResult {
+    /// Public key
+    pub pubkey: PublicKey,
+    /// Label tweak
+    pub label_tweak: [u8; 32],
+}
+
+/// Label tweak errors
+#[derive(Debug, Clone, Copy, Eq, PartialEq, PartialOrd, Ord, Hash)]
+pub enum LabelTweakError {
+    /// Unexpected failures
+    Failure,
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for LabelTweakError {}
+
+impl fmt::Display for LabelTweakError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match self {
+            LabelTweakError::Failure => write!(f, "Failed to create label tweak"),
+        }
+    }
+}
+
+/// Create Silent Payment label tweak and label.
+pub fn silentpayments_recipient_create_label_tweak<C: Verification>(
+    secp: &Secp256k1<C>,
+    recipient_scan_key: &SecretKey,
+    m: u32,
+) -> Result<LabelTweakResult, LabelTweakError> {
+
+    let cx = secp.ctx().as_ptr();
+    unsafe {
+
+        let mut pubkey = ffi::PublicKey::new();
+        let mut label_tweak32 = [0u8; 32];
+
+        let res = ffi::secp256k1_silentpayments_recipient_create_label_tweak(
+            cx,
+            &mut pubkey,
+            label_tweak32.as_mut_c_ptr(),
+            recipient_scan_key.as_c_ptr(),
+            m,
+        );
+
+        if res == 1 {
+            let pubkey = PublicKey::from(pubkey);
+            let label_tweak = label_tweak32;
+
+            return Ok(LabelTweakResult { pubkey, label_tweak });
+        } else {
+            return Err(LabelTweakError::Failure);
+        }
```

Comment:
Syntax could be more "rusty" by removing the 'return' keywords (this is a clippy warning, see other clippy warnings for more simple improvements like this)

Reply:
Thanks. Done in 6fbb0595e36f9acf4f24e20e43c7c790ee91f22e